client/src/components/Applications/AppForm/AppForm.jsx

```diff
@@ -459,17 +463,37 @@ export const AppSchemaForm = ({ app }) => {
         onSubmit={(values, { setSubmitting, resetForm }) => {
           const job = cloneDeep(values);
 
-          job.fileInputs = Object.entries(job.fileInputs)
-            .map(([k, v]) => {
-              // filter out read only inputs. 'FIXED' inputs are tracked as readOnly
-              if (
-                Object.hasOwn(appFields.fileInputs, k) &&
-                appFields.fileInputs[k].readOnly
-              )
-                return;
-              return { name: k, sourceUrl: v };
-            })
-            .filter((fileInput) => fileInput && fileInput.sourceUrl); // filter out any empty values
+          // Transform input field values into format that jobs service wants
+          job.fileInputs = Object.values(
+            Object.entries(job.fileInputs)
+              .map(([k, v]) => {
+                // filter out read only inputs. 'FIXED' inputs are tracked as readOnly
+                if (
+                  Object.hasOwn(appFields.fileInputs, k) &&
+                  appFields.fileInputs[k].readOnly
+                )
+                  return;
+                return {
+                  name: k,
+                  sourceUrl: !isTargetPathField(k) ? v : null,
+                  targetDir: isTargetPathField(k) ? v : null,
+                };
+              })
+              .filter((fileInput) => fileInput.sourceUrl || fileInput.targetDir) // filter out any empty values
```

Comment:
This will pass through a fileInput as long as either the sourceUrl or targetDir is defined, but I think we'll need to filter out fileInput objects that do not include a sourceUrl. Otherwise Tapis will respond with an error if a fileInput with a targetDir is defined, but has no sourceUrl

Reply:
Thanks @rstijerina  for catching this issue. I fixed and tested again